### PR TITLE
Export spreadsheet

### PR DESCRIPTION
**Pull request by jerryfreshworks**
image

**Resolves:**

- export adm322 spreadsheet

**Changes:**

- backend: read adm322 and fill the value

- frontend: download spreadsheet

- format of file name: `booking_[interpreter first name]_[booking first date].xlsx`

**References:**

- [CSBISS-51](https://justice.gov.bc.ca/jira/browse/CSBISS-51)

### Prerequisites

- [x] My code follows the code style of this project. (https://sites.google.com/freshworks.io/virtual-clinic/technical-overview/coding-guidelines-dev-team-agreement and https://sites.google.com/freshworks.io/virtual-clinic/technical-overview/coding-guidelines-dev-team-agreement/full-stack)
- [x] I have considered the maintainability of the code I have written
- [ ] I have updated the documentation according to my changes.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: frontend/src/index.tsx [hunk 1]
```diff
@@ -71,6 +71,9 @@ const Routes = () => {
 
 const Start: React.FC = () => {
   const [{ data, error, loading }] = useAxiosGet('/config');
+  // for debug the login issue on TEST env.
+  console.log(data, error, loading);
+
```
Comment (jerryfreshworks) [inline frontend/src/index.tsx:76]: This is going to debug the TEST env login issue(redirect to wrong KC). I will remove this once we see the console log.

**Review by will-freshworks — CHANGES_REQUESTED** [review]
Minor refactoring changes. Logic looks good though.

File: api/src/booking/booking.controller.ts [hunk 2]
```diff
@@ -102,4 +99,119 @@ export class BookingController {
...
+  @Get('/export/:id')
+  @Header('Content-Type', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet')
+  @Header('Content-Disposition', 'attachment; filename=test.xlsx')
+  async exportExcel(@Param('id') id: string, @Res() res: Response) {
+    // query booking
+    const booking = await this.bookingService.findOne(+id);
+    const { interpreter } = booking;
+
+    // generate excel file
+    const workbook = new ExcelJS.Workbook();
```
[the remark this answers is not in the document]
Comment (jerryfreshworks) [reply api/src/booking/booking.controller.ts:112]: Thanks for catching this.

File: frontend/src/index.tsx [hunk 3]
```diff
@@ -71,6 +71,9 @@ const Routes = () => {
 
 const Start: React.FC = () => {
   const [{ data, error, loading }] = useAxiosGet('/config');
+  // for debug the login issue on TEST env.
+  console.log(data, error, loading);
```
[the remark this answers is not in the document]
Comment (jerryfreshworks) [reply frontend/src/index.tsx:75]: ya, I think this is one line change, so I want to deploy this with this PR.
I'm guessing this causing the issue, so I feel this console log might help me to figure the weird login issue on TEST env.

[the remark this answers is not in the document]
Comment (will-freshworks) [reply frontend/src/index.tsx:75]: The more logs the better. Could you use `console.info` though in that case?

**Review by will-freshworks — APPROVED** [review]
Nice :)

File: frontend/src/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jerryfreshworks) [reply frontend/src/index.tsx]: ya, adding more logs.

**Pull request merged**